api/layer/layer.go

```diff
@@ -569,7 +569,7 @@ func (n *layer) deleteObject(ctx context.Context, bkt *data.BucketInfo, settings
 		}
 
 		if obj.DeleteMarkVersion, obj.Error = n.removeOldVersion(ctx, bkt, nodeVersion, obj); obj.Error != nil {
-			return obj
+			n.log.Error("remove old version", zap.Error(obj.Error))
```

Comment:
Why does it happen? I'm not sure it's "Error" level, maybe it's perfectly fine in some cases.

Reply:
I didn't find the exact reason for the problem, but it is directly connected to the multithreading in the test. Without threads, the code works perfectly. I consider that in some cases not all objects are removed because some error handling returns as here. Well, the `error` maybe too much, replaces with `Info`

Reply:
@smallhive do we know exact `obj.Error` which caused test failures? May be we can exclude only "normal" errors here

Reply:
I tried this approach. The error in this place is Object not found and sometimes Object already removed or something like that.
In this line we are trying to remove some _old_ version. Returning from here allows for the  _actual_ version be not removed, and this fix removes all versions.
Anyway, this line fixes the tests and don't mark any other tests with red